docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+:::note
+The operator will be installed in a new namespace called "my-percona-server-mongodb-operator" and will be usable from that namespace only.
+:::
+3. Create a mongodb service.
+
+import CreateMongodbService from './_create-mongodb-service.mdx';
+
+<CreateMongodbService/>
+
+### Implement the code logic
+:::note
+If you're already running `odo dev` in a terminal, exit it and start afresh.
+:::
+
+The new code is simple. We obtain the connection information (username, password, and host) from the environment, and use it to connect to the mongodb service and ping it. 
+
+Replace the content of your `main.go` with the following content:
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// Connection URI
+var (
+	USERNAME = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_USER")
+	PASSWORD = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_PASSWORD")
+	HOST     = os.Getenv("PERCONASERVERMONGODB_HOST")
+	uri      = fmt.Sprintf("mongodb://%s:%s@%s:27017/?maxPoolSize=20&w=majority", USERNAME, PASSWORD, HOST)
+)
+
+func main() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe("0.0.0.0:8080", nil)
+}
+
+func HelloServer(w http.ResponseWriter, r *http.Request) {
+	// Create a new client and connect to the server
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		fmt.Fprintf(w, "failed to connect: %s", err.Error())
+		return
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			fmt.Fprintf(w, "failed to disconnect: %s", err.Error())
+			return
+		}
+	}()
+
+	// Ping the primary
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		fmt.Fprintf(w, "unable to connect to the server: %s", err.Error())
+		return
+	}
+
+	fmt.Fprintf(w, "Successfully connected and pinged.")
+}
+```
+We will be using the mongodb client library, so let's update the go.mod with this dependency by running the following command: 
+```shell
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+### Run the application on the cluster
+Run this application on the cluster with `odo dev`.
+```shell
+odo dev
+```
+```shell
+$ odo dev
+  __
+ /  \__     Developing using the my-go-app Devfile
+ \__/  \    Namespace: my-percona-server-mongodb-operator
+ /  \__/    odo version: v3.0.0-rc1
+ \__/
+
+↪ Deploying to the cluster in developer mode
+ •  Waiting for Kubernetes resources  ...
+ ⚠  Pod is Pending
+ ✓  Pod is Running
+ ✓  Syncing files into the container [152ms]
+ ✓  Building your application in container on cluster (command: build) [15s]
+ •  Executing the application (command: run)  ...
+ -  Forwarding from 127.0.0.1:40001 -> 8080
+
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
```

Comment:
remove spacing? I feel like a lot of these console outputs have a lot of spacing already